publications/2023-neurips/lcdb/workflow/keras/_dense.py

```diff
@@ -358,17 +441,36 @@ def _predict(self, X):
         y_pred = self._transformer_label.inverse_transform(y_pred)
         return y_pred
 
-    def _predict_proba(self, X):
-        X = self.transform(X, y=None, metadata=self.metadata).astype(np.float32)
-        y_pred = self.learner.predict(
-            X, batch_size=min(len(X), self.batch_size), verbose=self.verbose
-        )
-        return y_pred
+    def _predict_proba(self, X, apply_transform=True, use_snapshot_ensemble=None):
```

Comment:
This is a tricky one. This is because when making a normal prediction with `predict_proba`, e.g., during training, the snapshots should not be used. So this is not a property only of the network itself but also of how predictions should be made. Or would you suggest another way of doing this?

Reply:
I am not sure to understand what is "a normal prediction. If snapshot ensemble is used, the resulting model is the ensemble of snapshots? So, I thought the `predict_proba` should be the mixture distribution of ensemble members.